bundle/edu.gemini.itc.web/src/main/scala/edu/gemini/itc/web/ITCRequest.scala

```diff
@@ -275,19 +275,37 @@ object ITCRequest {
   def sourceDefinitionParameters(r: ITCRequest): SourceDefinition = {
     // Get the source geometry and type
     import SourceDefinition.Profile._
+
+    def magnitudeSystemFor(s: String) = s match {
+      case "MAG"                  => MagnitudeSystem.Vega
+      case "ABMAG"                => MagnitudeSystem.AB
+      case "JY"                   => MagnitudeSystem.Jy
+      case "WATTS"                => MagnitudeSystem.Watts
+      case "ERGS_WAVELENGTH"      => MagnitudeSystem.ErgsWavelength
+      case "ERGS_FREQUENCY"       => MagnitudeSystem.ErgsFrequency
```

Comment:
Well, I guess, you could argue if the exception that the `match` command in Scala throws isn't good enough. Don't know. Anyone has a strong opinion on that? (At least Scala *does* throw in these cases, which is great and often probably good enough.)